RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -210,6 +213,16 @@ void PhotonProducer::produce(edm::Event& theEvent, const edm::EventSetup& theEve
   }
   if( validEcalRecHits) endcapRecHits = *(endcapHitHandle.product());
 
+  bool validPreshowerRecHits=true;
+  Handle<EcalRecHitCollection> preshowerHitHandle;
+  theEvent.getByToken(preshowerHits_, preshowerHitHandle);
+  EcalRecHitCollection preshowerRecHits;
+  if (!preshowerHitHandle.isValid()) {
+    throw cms::Exception("PhotonProducer") 
+      << "Error! Can't get the preshowerEcalHits";
+  }
```

Comment:
A class must have consistent behavour with itself at the minimum. Either all similar missing inputs throw or none do. Given that in this producer they the rec hits dont throw, then lets keep it that way. Also less a of a pain for anybody actually using the class still.